### PR TITLE
TransfQMix release

### PR DESCRIPTION
**Pull request by mttga**

### CODE REVIEW

**Review by amacrutherford — COMMENTED** [review]
looks cool! couple of minor points, also would be interesting to see the SMAX results

File: jaxmarl/wrappers/baselines.py
[the remark this answers is not in the document]
Comment (mttga) [reply jaxmarl/wrappers/baselines.py:1]: oh yes true, will do

File: jaxmarl/wrappers/transformers.py
[the remark this answers is not in the document]
Comment (mttga) [reply jaxmarl/wrappers/transformers.py:1]: yes every env that wants to use trasformers will need to wrap the environment observations in some way to be readable by the transformer. I could also reorganize it to have per-env specific wrappers if that would look better. I had put a note on the readme but now I see I deleted by mistake

File: baselines/QLearning/utils/fast_attention.py
[the remark this answers is not in the document]
Comment (mttga) [reply baselines/QLearning/utils/fast_attention.py:1]: the multihead attention from the fastattention script is significantly faster and more stable than the default flax one, which is beneficial in RL.

The script was taken in https://github.com/google-research/google-research/blob/master/performer/fast_attention/jax/fast_attention.py

Notice that the use of fast attention is optional: https://github.com/FLAIROx/JaxMARL/blob/4d78674dc5195683f3cd6e7e9d6799ddd586a714/baselines/QLearning/transf_qmix.py#L51

File: baselines/QLearning/transf_qmix.py [hunk 1]
```diff
@@ -0,0 +1,1054 @@
...
+        )
+        runner_state, metrics = jax.lax.scan(
+            _update_step, runner_state, None, config["NUM_UPDATES"]
+        )
+        return {'runner_state':runner_state, 'metrics':metrics}
+    
+    return train
+
+
+def signle_run(config):
```
[the remark this answers is not in the document]
Comment (mttga) [reply baselines/QLearning/transf_qmix.py:931]: yes 🙃 thanks

**Comment by mttga** [comment]
Here are the results average across 4 seeds:
- test won: https://api.wandb.ai/links/mttga/2beham1m
- test returns: https://api.wandb.ai/links/mttga/6yszv25o

Results are better than QMix for most of the maps with except of 3s5z and 3s5z_vs_3s_6z. But the main advantage of transformers is the potential transferability of the agent parameters and the learned qmix function between scenarios.

File: jaxmarl/wrappers/baselines.py
Comment (amacrutherford) [inline jaxmarl/wrappers/baselines.py:1]: why we removing the hanabi option?

**Comment by amacrutherford** [comment]
> Here are the results average across 4 seeds:
> 
>     * test won: https://api.wandb.ai/links/mttga/2beham1m
> 
>     * test returns: https://api.wandb.ai/links/mttga/6yszv25o
> 
> 
> Results are better than QMix for most of the maps with except of 3s5z and 3s5z_vs_3s_6z. But the main advantage of transformers is the potential transferability of the agent parameters and the learned qmix function between scenarios.

awesome!

File: jaxmarl/wrappers/baselines.py
Comment (amacrutherford) [reply jaxmarl/wrappers/baselines.py:1]: cools !
